apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/builder/ReferenceAnalyzer.java

```diff
@@ -230,10 +230,14 @@ void extractReferences(IApiTypeContainer scope, IProgressMonitor monitor) throws
 	 */
 	public IApiProblem[] analyze(IApiComponent component, IApiTypeContainer scope, IProgressMonitor monitor) throws CoreException {
 		SubMonitor localMonitor = SubMonitor.convert(monitor, 4);
-		// build problem detectors
-		IApiProblemDetector[] detectors = buildProblemDetectors(component, ProblemDetectorBuilder.K_ALL, localMonitor.split(1));
-		// analyze
 		try {
+			// build problem detectors
+			IApiProblemDetector[] detectors = buildProblemDetectors(component, ProblemDetectorBuilder.K_ALL,
```

Comment:
Is there a reason why you moved this into the try-catch-block?

Reply:
sure: the final block cleans up a field that is initialized somewhen within  "buildProblemDetectors". Any exception after initializing the field would leave a leak.